### PR TITLE
Linkify cross-repo issues references in titles - Fixes #412

### LINKED ISSUE #412: Cross-repo issue link in issue title
See https://github.com/rust-lang/rust/pull/41830. The link in the title goes to `#4000` of the current repo even though it should be possible to match the `org/repo#num` pattern and make the correct link.

### PR DESCRIPTION
**Pull request by durka**
First stab at fixing #412.

before: 
after:

### CODE REVIEW

**Comment by durka** [comment]
Travis failed but it doesn't say why. How can I run the tests?

File: extension/util.js [hunk 1]
```diff
@@ -31,4 +31,13 @@
 
 		return success;
 	};
+
+    exports.issueRegex = /([a-zA-Z0-9-_.]+\/[a-zA-Z0-9-_.]+)?#[0-9]+/;
+	exports.linkifyIssue = (repoPath, issue, attrs) => {
```
Comment (sindresorhus) [inline extension/util.js]: I would call it `linkifyIssueReference` or `linkifyIssueRef`

File: extension/util.js [hunk 2]
```diff
@@ -31,4 +31,13 @@
 
 		return success;
 	};
+
+    exports.issueRegex = /([a-zA-Z0-9-_.]+\/[a-zA-Z0-9-_.]+)?#[0-9]+/;
```
Comment (sindresorhus) [inline extension/util.js]: Incorrect indentation here.

File: extension/content.js [hunk 3]
```diff
@@ -148,11 +148,12 @@ function addDeleteForkLink() {
...
 	const $title = $('.js-issue-title');
 	const titleText = $title.text();
+    const issueRegex = utils.issueRegex;
 
-	if (/(#\d+)/.test(titleText)) {
+	if (issueRegex.test(titleText)) {
 		$title.html(titleText.replace(
-			/#(\d+)/g,
-			`<a href="https://github.com/${repoUrl}/issues/$1">#$1</a>`
+			RegExp(issueRegex.source, 'g'),
```
Comment (sindresorhus) [inline extension/content.js]: `RegExp` => `new RegExp`

File: extension/content.js [hunk 4]
```diff
@@ -148,11 +148,12 @@ function addDeleteForkLink() {
 function linkifyIssuesInTitles() {
 	const $title = $('.js-issue-title');
 	const titleText = $title.text();
+    const issueRegex = utils.issueRegex;
```
Comment (sindresorhus) [inline extension/content.js]: Incorrect indentation.

**Comment by sindresorhus** [comment]
@durka It does say why. Look at the Travis log: https://travis-ci.org/sindresorhus/refined-github/builds/231018899

**Comment by durka** [comment]
Strange. When I looked at the Travis log before it was blank. Anyway, I fixed the tidy issues but I do not understand the [remaining error](https://travis-ci.org/sindresorhus/refined-github/builds/231240558#L708) saying `utils` is undefined. As far as I can tell I used `utils` in `linkify-urls-in-code.js` in exactly the same way it's used in other files.

**Comment by jgierer12** [comment]
@durka that's ava complaining since it doesn't know about `utils.js`. @sindresorhus is there any way to mock/import `utils.js` or simulate a chrome extension environment?

File: extension/content.js (hunk 3 above)
Comment (durka) [inline extension/content.js]: Is there any difference?

Comment (sindresorhus) [inline extension/content.js]: Not semantically. Just a preference and for consistency. You always use `new` for your own constructors. `RegExp` without `new` is just a quirk in the language.

**Comment by sindresorhus** [comment]
Looks good. Thanks @durka :)

**Comment by sindresorhus** [comment]
@jgierer12 Yes, I fixed it on merge.

**Pull request merged**